src/app/state/importaccounts/saga.ts

```diff
@@ -71,33 +78,64 @@ function* enumerateAccountsFromMnemonic(action: PayloadAction<string>) {
   }
 }
 
+function* fetchBalanceForAccount(account: ImportAccountsListAccount) {
+  let currentStep = (yield* select(selectImportAccounts)).step
+  if (currentStep === ImportAccountsStep.Idle) {
+    yield* setStep(ImportAccountsStep.LoadingBalances)
+  }
+  const balance = yield* call(getBalance, hex2uint(account.publicKey))
+  yield* put(
+    importAccountsActions.updateAccountBalance({
+      address: account.address,
+      balance,
+    }),
+  )
+  currentStep = (yield* select(selectImportAccounts)).step
+  const hasMissingBalances = yield* select(selectImportAccountHasMissingBalances)
+  if (currentStep === ImportAccountsStep.LoadingBalances && !hasMissingBalances) {
+    yield* setStep(ImportAccountsStep.Idle)
+  }
+}
+
+function* ensureAllBalancesArePresentOnCurrentPage() {
+  const existingAccounts = yield* select(selectImportAccountsOnCurrentPage)
+  const accountsWithNoBalance = existingAccounts.filter(a => !a.balance)
+  for (let i = 0; i < accountsWithNoBalance.length; i++) {
+    yield* fork(fetchBalanceForAccount, accountsWithNoBalance[i])
+  }
```

Comment:
Does this intentionally not wait? I'd expect `yield* all([ ... ])`

Reply:
Re-implemented using `yield* all()`. (The flow was the same both before and after the change.)